### PR TITLE
Add explicit auth option for AzureCli to better support Federated Cre…

### PR DESCRIPTION
**Pull request by bbowman**
…dential scenarios / situations where AzCli is preferred but some other auth in the default cred fallbacks is being used instead

### CODE REVIEW

**Comment by bbowman** [comment]
@alberto-baron you seem most active here, any chance you could help me on this?

File: src/PackageUploader.ClientApi/Client/Ingestion/TokenProvider/AzureCliAccessTokenProvider.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    public AzureCliAccessTokenProvider(IOptions<AccessTokenProviderConfig> config, ILogger<AzureCliAccessTokenProvider> logger)
+    {
+        _config = config.Value;
+        _logger = logger;
+    }
+
+    public async Task<IngestionAccessToken> GetTokenAsync(CancellationToken ct)
+    {
+        var azureCredentialOptions = new DefaultAzureCredentialOptions
```
[the remark this answers is not in the document]
Comment (bbowman) [reply src/PackageUploader.ClientApi/Client/Ingestion/TokenProvider/AzureCliAccessTokenProvider.cs:29]: @alberto-baron great suggestion, went ahead and switched over. I need to also re-confirm this works in our use case but pushed the update for you to review.

**Comment by bbowman** [comment]
@alberto-baron can you help get this over the line?

File: src/PackageUploader.ClientApi/Client/Ingestion/TokenProvider/AzureCliAccessTokenProvider.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (alberto-baron) [reply src/PackageUploader.ClientApi/Client/Ingestion/TokenProvider/AzureCliAccessTokenProvider.cs:29]: I can't see any code change in the PR :o

[the remark this answers is not in the document]
Comment (bbowman) [reply src/PackageUploader.ClientApi/Client/Ingestion/TokenProvider/AzureCliAccessTokenProvider.cs]: @alberto-baron sorry about that! Helps to actually hit save sometimes!

[the remark this answers is not in the document]
Comment (bbowman) [reply src/PackageUploader.ClientApi/Client/Ingestion/TokenProvider/AzureCliAccessTokenProvider.cs]: You should see the changes now

**Review by alberto-baron — APPROVED** [review]

**Pull request merged**